### PR TITLE
Benchmarks and baseline profile generation

### PR DESCRIPTION
**Pull request by mlykotom**
```
StartupBenchmarks_startupCompilationWarmup
     timeToFullDisplayMs   min 308.0,   median 314.4,   max 370.4
  timeToInitialDisplayMs   min 289.3,   median 297.9,   max 352.0
    Traces: Iteration 0 1 2 3 4

StartupBenchmarks_startupCompilationPartial
     timeToFullDisplayMs   min 230.3,   median 247.0,   max 272.4
  timeToInitialDisplayMs   min 221.9,   median 228.0,   max 257.6
    Traces: Iteration 0 1 2 3 4

StartupBenchmarks_startupCompilationNone
     timeToFullDisplayMs   min 274.6,   median 278.1,   max 285.3
  timeToInitialDisplayMs   min 258.4,   median 262.3,   max 265.9
    Traces: Iteration 0 1 2 3 4
```

### CODE REVIEW

File: macrobenchmark/src/main/java/com/google/samples/apps/sunflower/macrobenchmark/PlantListBenchmarks.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+fun MacrobenchmarkScope.goToPlantListTab() {
+    // find the tab with plants list
+    val plantListTab = device.findObject(By.descContains("Plant list"))
+    plantListTab.click()
+    // wait until idle
+    device.waitForIdle()
+    // sleep for animations to settle
+    Thread.sleep(500)
```
Comment (keyboardsurfer) [inline macrobenchmark/src/main/java/com/google/samples/apps/sunflower/macrobenchmark/PlantListBenchmarks.kt:68]: How is this sleep necessary? Can we avoid it using a listener once the animation is finished?

File: build.gradle [hunk 2]
```diff
@@ -18,38 +18,43 @@ buildscript {
...
+        truthVersion = '1.1.3'
+        testExtJunit = '1.1.3'
         uiAutomatorVersion = '2.2.0'
         viewPagerVersion = '1.0.0'
-        workVersion = '2.4.0'
+        workVersion = '2.7.1'
+
+        profileInstallerVersion = '1.2.0-alpha02'
+        uiAutomatorVersion = '2.2.0'
+        benchmarkVersion = '1.1.0-rc01'
```
Comment (keyboardsurfer) [inline build.gradle:56]: rc02

File: build.gradle [hunk 3]
```diff
@@ -18,38 +18,43 @@ buildscript {
...
-        truthVersion = '0.42'
-        testExtJunit = '1.1.0'
+        truthVersion = '1.1.3'
+        testExtJunit = '1.1.3'
         uiAutomatorVersion = '2.2.0'
         viewPagerVersion = '1.0.0'
-        workVersion = '2.4.0'
+        workVersion = '2.7.1'
+
+        profileInstallerVersion = '1.2.0-alpha02'
```
Comment (keyboardsurfer) [inline build.gradle:54]: beta01

**Review by keyboardsurfer — APPROVED** [review]
Approved with version bump request.

File: macrobenchmark/src/main/java/com/google/samples/apps/sunflower/macrobenchmark/PlantListBenchmarks.kt (hunk 1 above)
Comment (mlykotom) [reply macrobenchmark/src/main/java/com/google/samples/apps/sunflower/macrobenchmark/PlantListBenchmarks.kt]: Some benchmarks finished earlier than the actual content was loaded. Changed it with waiting until Recycler has children.

**Pull request merged**